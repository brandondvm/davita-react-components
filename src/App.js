import Button from './components/Button/Button';
import Checkbox from './components/Checkbox/Checkbox';
import Counter from './components/Counter/Counter';
import Input from './components/Input/Input';
import RadioButtonGroup from './components/RadioButtonGroup/RadioButtonGroup';
import SegmentedController from './components/SegmentedController/SegmentedController';
import Stepper from './components/Stepper/Stepper';
import ReadOnlyInput from './components/ReadOnlyInput/ReadOnlyInput';

import './styles/styles.scss';

const radioGroupItems = [
  { id: 'radioOption1', label: 'Option 1', name: 'radiobuttongroup', value: 1, checked: true },
  { id: 'radioOption2', label: 'Option 2', name: 'radiobuttongroup', value: 2, checked: false },
  { id: 'radioOption3', label: 'Option 3', name: 'radiobuttongroup', value: 3, checked: false },
];

const segmentedControllerItems = [
  { id: "option1", label: "option 1", name: "segmentedControllerOption", value: 1, checked: true },
  { id: "option2", label: "option 2", name: "segmentedControllerOption", value: 2, checked: false },
  { id: "option3", label: "option 3", name: "segmentedControllerOption", value: 3, checked: false }
];

const handleRadioButtonClick = ({ value }) => {
  console.log('Radio Button: ', value);
}

const handleCheckboxClick = ({ label, isSelected }) => {
  console.log(`Checkbox ${label}: `, isSelected);
};

const handleButtonClick = () => {
  this.props.history.push('/');
};

function App() {
  return (
    <div className="app-wrapper">
      <Stepper steps={ 6 } />
      <Input id="empty-field" placeholder="Empty field"/>
      <Input selectButtons={{
          leftButtonLabel: 'lbs',
          leftButtonOnClick: () => {},
          rightButtonLabel: 'kg',
          rightButtonOnClick: () => {}
        }}
        placeholder="Placeholder"
        label="Label"
        id="select-buttons"
      />
      <br />
      <Input id="input-error" errorMessage="Error message" placeholder="Placeholder" label="Label"/>
      <Input id="inner-label" innerLabel="Units" placeholder="Placeholder" label="Label"/>
      <Input id="char-max" characterMax="20" placeholder="Placeholder" label="Label"/>
      <ReadOnlyInput value="Read only text" label="Read Only"/>
      <SegmentedController controllerOptions={ segmentedControllerItems } />
      <Checkbox id="checkbox1" name="checkbox1" checked label="Option 1" onChange={ handleCheckboxClick } />
      <Checkbox id="checkbox2" name="checkbox2" label="Option 2" onChange={ handleCheckboxClick } />
      <RadioButtonGroup options={ radioGroupItems } onItemClick={ handleRadioButtonClick } register={() => {}} />
      <Counter label="1.5% Yellow Bag" />
      <br />
      <Button label="Next" onClick={ handleButtonClick } />
      <Button label="Disabled" disabled={ true } />
    </div>
  );
}

export default App;
