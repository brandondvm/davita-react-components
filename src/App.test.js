import enzyme from 'enzyme';
import React from 'react';

import App from './App';

function setup() {
  return enzyme.shallow(<App />);
}

describe('App Component', () => {
  test('Component renders correctly', () => {
    const wrapper = setup();
    expect(wrapper.get(0)).toMatchSnapshot();
  });
});
