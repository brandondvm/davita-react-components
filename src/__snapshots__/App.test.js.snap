// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Component renders correctly 1`] = `
<div
  className="app-wrapper"
>
  <Stepper
    steps={6}
  />
  <Input
    id="empty-field"
    placeholder="Empty field"
  />
  <Input
    id="select-buttons"
    label="Label"
    placeholder="Placeholder"
    selectButtons={
      Object {
        "leftButtonLabel": "lbs",
        "leftButtonOnClick": [Function],
        "rightButtonLabel": "kg",
        "rightButtonOnClick": [Function],
      }
    }
  />
  <br />
  <Input
    errorMessage="Error message"
    id="input-error"
    label="Label"
    placeholder="Placeholder"
  />
  <Input
    id="inner-label"
    innerLabel="Units"
    label="Label"
    placeholder="Placeholder"
  />
  <Input
    characterMax="20"
    id="char-max"
    label="Label"
    placeholder="Placeholder"
  />
  <ReadOnlyInput
    label="Read Only"
    value="Read only text"
  />
  <SegmentedController
    controllerOptions={
      Array [
        Object {
          "checked": true,
          "id": "option1",
          "label": "option 1",
          "name": "segmentedControllerOption",
          "value": 1,
        },
        Object {
          "checked": false,
          "id": "option2",
          "label": "option 2",
          "name": "segmentedControllerOption",
          "value": 2,
        },
        Object {
          "checked": false,
          "id": "option3",
          "label": "option 3",
          "name": "segmentedControllerOption",
          "value": 3,
        },
      ]
    }
  />
  <Checkbox
    checked={true}
    id="checkbox1"
    label="Option 1"
    name="checkbox1"
    onChange={[Function]}
  />
  <Checkbox
    id="checkbox2"
    label="Option 2"
    name="checkbox2"
    onChange={[Function]}
  />
  <RadioButtonGroup
    onItemClick={[Function]}
    options={
      Array [
        Object {
          "checked": true,
          "id": "radioOption1",
          "label": "Option 1",
          "name": "radiobuttongroup",
          "value": 1,
        },
        Object {
          "checked": false,
          "id": "radioOption2",
          "label": "Option 2",
          "name": "radiobuttongroup",
          "value": 2,
        },
        Object {
          "checked": false,
          "id": "radioOption3",
          "label": "Option 3",
          "name": "radiobuttongroup",
          "value": 3,
        },
      ]
    }
    register={[Function]}
  />
  <Counter
    label="1.5% Yellow Bag"
  />
  <br />
  <Button
    label="Next"
    onClick={[Function]}
  />
  <Button
    disabled={true}
    label="Disabled"
  />
</div>
`;
